lib/customer.rb

```diff
@@ -0,0 +1,50 @@
+require 'csv'
+
+class Customer
+  attr_reader :id
+  attr_accessor :email, :address
+
+  def initialize(id, email, address)
+    @id = id
+    @email = email
+    @address = address
+  end
+
+  def self.all
+    all_customers = CSV.read('data/customers.csv').map { |row| row.to_a }
+
+    all_customers = all_customers.map do |customer|
+      id = customer[0].to_i
+      email = customer[1]
+      address = {
+          street: customer[2],
+          city: customer[3],
+          state: customer[4],
+          zip: customer[5]
+      }
+      Customer.new(id, email, address)
+    end
+
+    return all_customers
+  end
+
+  def self.find(id)
+    customer_found = self.all.select { |customer| customer.id == id }
+    customer_found.empty? ? (return nil) : (return customer_found[0])
```

Comment:
I like your thinking here-- I like using `select`, and if we get an empty array back, we return `nil`, otherwise we return the first one in the list.

IT JUST SO HAPPENS that [the Enumerable method `find`](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-find) happens to:
- return after finding the first match
- returns `nil` if no match was found

... which is really the logic you just wrote!

I don't think it's necessary to refactor this method, but I just thought it'd be good to know that `find` happens to perfectly match the logic you wrote. :)

Reply:
Oh!  Wow that's smart - thank you for pointing that out. :)